src/scripts/website-blocking/WebsiteBlocking.ts

```diff
@@ -117,8 +121,33 @@ class WebsiteBlocking {
     }
     return null;
   }
-}
 
-export default WebsiteBlocking;
+  static async closeOnStart() {
+    
+    const websiteBlockingList = JSON.parse(await db().get('websiteBlockingList') || '[]');
 
+    const blockedSites = websiteBlockingList.map(entry => entry.url);
 
+    console.log(blockedSites)
+    if (blockedSites.length > 0) {
+      const blockedTabs = await browser.tabs.query({ url: blockedSites });
+      blockedTabs.forEach(tab => youShallNotPass(tab.id));
+    }
+  };
+
+  static async blockSite(tabId, changeInfo) {
+    const url = changeInfo.pendingUrl || changeInfo.url;
+    
```

Comment:
Let's put early return here if `url` is falsy.
This function seems to be triggered a lot - e.g. when title is changed on a tab.